pkg/server/endpoints/registration/handler.go

```diff
@@ -740,3 +751,45 @@ func getCallerID(ctx context.Context) string {
 	callerID, _ := ctx.Value(callerIDKey{}).(string)
 	return callerID
 }
+
+func validateDNS(dns string) error {
+	// follow https://tools.ietf.org/html/rfc5280#section-4.2.1.6
+	// do not allow empty or the technically valid DNS " "
+	dns = strings.TrimSpace(dns)
+	if len(dns) == 0 {
+		return errors.New("dns is empty or only whitespace")
```

Comment:
If we update error message similar to comment above, we can probably drop the "dns" prefixing on all these error messages